hyper/contrib.py

```diff
@@ -43,28 +48,40 @@ def get_connection(self, host, port, scheme, cert=None):
         if cert is not None:
             ssl_context = init_context(cert=cert)
 
+        connection_key = (host, port, scheme, cert, proxy)
```

Comment:
Calling `get_connection` with the same arguments but different `proxy_headers` would reuse already created connection, so the new `proxy_headers` will be ignored.

Our `proxy_headers` are generated from proxy url, so we should be fine, but this is not strictly guaranteed in this code.

Maybe we should move in the `select_proxy` logic here, thus generating `proxy_headers` inside this function?